src/builder/QuarkBuilder.sol

```diff
@@ -332,6 +332,166 @@ contract QuarkBuilder {
+                Actions.constructBridgeOperations(
+                    Actions.BridgeOperationInfo({
+                        assetSymbol: payment.currency,
+                        amountNeededOnDst: maxCostOnDstChain,
+                        dstChainId: swapIntent.chainId,
+                        recipient: swapIntent.sender,
+                        blockTimestamp: swapIntent.blockTimestamp,
+                        // TODO: set
+                        useQuotecall: false
+                    }),
+                    chainAccountsList,
+                    payment
+                );
+
+                for (uint256 i = 0; i < bridgeQuarkOperations.length; ++i) {
+                    quarkOperations[actionIndex] = bridgeQuarkOperations[i];
+                    actions[actionIndex] = bridgeActions[i];
+                    actionIndex++;
+                }
+            }
+        }
+
+        // Then, swap `amount` of `assetSymbol` to `recipient`
+        (quarkOperations[actionIndex], actions[actionIndex]) = Actions.matchaSwap(
+            Actions.MatchaSwap({
+                chainAccountsList: chainAccountsList,
+                entryPoint: swapIntent.entryPoint,
+                swapData: swapIntent.swapData,
+                sellToken: swapIntent.sellToken,
+                sellTokenSymbol: sellTokenSymbol,
+                sellAmount: swapIntent.sellAmount,
+                buyToken: swapIntent.buyToken,
+                buyTokenSymbol: buyTokenSymbol,
+                expectedBuyAmount: swapIntent.expectedBuyAmount,
+                chainId: swapIntent.chainId,
+                sender: swapIntent.sender,
+                blockTimestamp: swapIntent.blockTimestamp
+            }),
+            payment,
+            // TODO: Set this
+            false
+        );
+        actionIndex++;
+
+        // TODO: Merge transactions on same chain into Multicall. Maybe do that separately at the end via a helper function.
+
+        // Truncate actions and quark operations
+        actions = Actions.truncate(actions, actionIndex);
+        quarkOperations = Actions.truncate(quarkOperations, actionIndex);
+
+        // Validate generated actions for affordability
+        if (payment.isToken) {
+            assertSufficientPaymentTokenBalances(actions, chainAccountsList, swapIntent.chainId);
+        }
+
+        // Construct EIP712 digests
+        EIP712Helper.EIP712Data memory eip712Data;
+        if (quarkOperations.length == 1) {
+            eip712Data = EIP712Helper.EIP712Data({
+                digest: EIP712Helper.getDigestForQuarkOperation(
+                    quarkOperations[0], actions[0].quarkAccount, actions[0].chainId
+                    ),
+                domainSeparator: EIP712Helper.getDomainSeparator(actions[0].quarkAccount, actions[0].chainId),
+                hashStruct: EIP712Helper.getHashStructForQuarkOperation(quarkOperations[0])
+            });
+        } else if (quarkOperations.length > 1) {
```

Comment:
What about the case, FWIW, where `quarkOperations.length == 0`?

Reply:
In that case, EIP712Data will be an empty struct.